### DISCUSSION TITLE
mtg access returns wrong address

### DISCUSSION THREAD

**Discussion by vdrandom** [9seconds/mtg#192] [Q&A] [answered]
mtg seems to use the default ip for uris it returns when a user runs `mtg access /path/to/config.toml`. The thing is, it can be bound to any address present (e.g. localhost, for tcp proxy).

It's obviously a cosmetic thing, but it would be nice to be able to set external address / hostname in the settings instead of fixing uris manually. Especially since without it mtg tries to guess and may guess wrong.

**Comment by 9seconds** [comment, marked as answer]
Please check access' help :)

```console
$ mtg access --help
Usage: mtg access <config-path>

Print access information.

Arguments:
  <config-path>    Path to the configuration file.

Flags:
  -h, --help         Show context-sensitive help.
  -v, --version      Print version.

  -i, --ipv4=IP      Public IPv4 address for proxy. By default it is resolved via remote website
  -I, --ipv6=IP      Public IPv6 address for proxy. By default it is resolved via remote website
  -p, --port=UINT    Port number. Default port is taken from configuration file, bind-to parameter
  -x, --hex          Print secret in hex encoding.
```

`-i` and `-I` are literally these flags you are looking for.

**Comment by vdrandom** [comment]
Makes sense, thanks.

But it would be nice to have a config option for that as well.

**Comment by 9seconds** [comment]
But this value won't be used for a proxy itself. This is mostly to make and share a link to a friend :/ I'm still not really sure if such ad-hoc thing deserves an option in a file.

**Comment by 9seconds** [comment]
Since this is not an issue anymore, I'm converting this to discussion.
